index.js

```diff
@@ -27,11 +27,14 @@ module.exports = function setHeader(res, name, value) {
   //
   res.setHeader(name, value);
 
+  var symbols = (typeof Object.getOwnPropertySymbols == "function" ? Object.getOwnPropertySymbols(res) : []);
+  var symbol  = (symbols.length === 1 ? symbols[0] : "_headers");
```

Comment:
The assumption here is that `http.OutgoingMessage` has only one symbol property but this is not true. There is also `kIsCorked`. Currently `outHeadersKey` is added after `kIsCorked` and own keys are returned in the order they were added per spec so getting the last element in the array works.

```js
var symbol = symbols.length >= 1 ? symbols[symbols.length - 1] : '_headers';
```

If more symbols properties will be added another strategy must be found. For example,

```js
var symbol;
for (var i = 0; i < symbols.length; i++) {
  if (String(symbols[i] === 'Symbol(outHeadersKey)') {
    symbol = symbols[i];
    break;
  }
}
```

Reply:
You may notice I checked `symbols.length === 1` so it would only work on one symbol situation. I'll change to your suggestion.

Reply:
Yes, my point is that it doesn't work with current version of Node.js